### DISCUSSION TITLE
I write an application with 3 DXGI swap-chains, how to capture specific swap-chain?

### DISCUSSION THREAD

**Discussion by Demonese** [obsproject/obs-studio#11268] [Development Questions] [answered]
Does OBS provide relevant APIs to hide or select specific swap-chain?

**Comment by Hikari-Tsai** [comment, marked as answer]
OBS does not expose a public setting/API for selecting an arbitrary DXGI swap-chain inside one process. Game Capture injects its graphics hook into the target process and chooses a presented swap-chain using the hook's internal window/swap-chain matching logic. The source configuration selects the target window/process, not a swap-chain index.

If the three swap-chains belong to three distinct top-level windows, create separate Game Capture sources and target each window. That is the supported distinction OBS can use.

If multiple swap-chains are presented by the same window/process, selecting one requires changing the injected `win-capture` graphics hook (for example, adding a stable selection criterion around the DXGI `Present` hooks) and then passing that choice through the shared hook configuration. A raw pointer or creation-order index is not a robust user-facing identifier because swap-chains can be destroyed and recreated during resize/fullscreen transitions.

For an application you control, a more reliable solution is to expose each desired render target through a separate window or publish the selected texture through a dedicated OBS source/plugin rather than trying to steer Game Capture's private swap-chain heuristic.
